mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp

```diff
@@ -2283,6 +2283,35 @@ LogicalResult IndexOp::verify() {
   return success();
 }
 
+OpFoldResult IndexOp::fold(FoldAdaptor adaptor) {
+  auto linalgOp = cast<LinalgOp>((*this)->getParentOp());
+  int64_t flatDimPos =
+      cast<AffineDimExpr>(linalgOp.getShapesToLoopsMap().getResult(getDim()))
```

Comment:
LinalgStructuredInterface does not require `getShapesToLoopsMap` is a permutation, that's just the default (and only implemented case today). Guard with dyn_cast.

Reply:
Also if you're calling `getShapesToLoopsMap` anyway, `getStaticLoopBounds` is the safer and easier way to do this anyway and replaces all of the logic below with just checking the shape of the dim in question.

Reply:
Thanks, this simplifies the code a lot